_modules/conditionals.md

```diff
@@ -6,3 +6,119 @@ short_title: Buttons and Switches
+    if (buttons_get(1)) {
+        leds_set_color(1, 255, 0, 0);
+    } else {
+        leds_set_color(1, 255, 0, 0);
+    }
+    delay(100);
+}
+leds_set_color(15, 255, 0, 0);
+```
+
+This code block only checks if Button1 is pressed in 0.1 second increments for 10 seconds. 0.1 seconds is faster than the average person can press the button, so it will work just fine, and once the `for` loop is done, you can continue writing your code as normal.
+
+Now let's talk about the `else if` block. If we want to turn LED1 red when only one button is pressed, yellow when two buttons are pressed, or green when all three buttons are pressed, we could use the following:
+
+```c
+while (true) {
+    if (buttons_get(1) && buttons_get(2) && buttons_get(3)) {
+        leds_set_color(1, 0, 255, 0);
+    } else if (buttons_get(1) && buttons_get(2)) {
+        leds_set_color(1, 255, 255, 0);
+    } else if (buttons_get(2) && buttons_get(3)) {
+        leds_set_color(1, 255, 255, 0);
+    } else if (buttons_get(1) && buttons_get(3)) {
+        leds_set_color(1, 255, 255, 0);
+    } else if (buttons_get(1)) {
+        leds_set_color(1, 255, 0, 0);
+    } else if (buttons_get(2)) {
+        leds_set_color(1, 255, 0, 0);
+    } else if (buttons_get(3)) {
+        leds_set_color(1, 255, 0, 0);
+    } else {
+        leds_set_color(1, 0, 0, 0);
+    }
+}
+```
+
+There are two important things to notice in this example:
+
+1. We can "chain" as many `else if` blocks together as we want! This gives us a lot of freedom to make our code do anything we want it to.
+
+1. We introduced the use of `&&` which is a "logical AND". Using `&&` we can link _two or more_ conditionals together and only execute the code inside the block if they are ALL `true`. For example, we can read `if (buttons_get(1) && buttons_get(2) && buttons_get(3)) {...` as "if button 1 is pressed AND button 2 is pressed AND button 3 is pressed..."
+
+Let's try one final example using a switch instead of a button:
+
+```c
+while (true) {
+    while (true) {
+        if (switches_get(2)) {
+            leds_set_color(1, 255, 0, 0);
+            break;
+        }
+    }
+    while (true) {
+        if (!switches_get(2)) {
+            leds_set_color(1, 0, 0, 0);
+            break;
+        }
+    }
+}
+```
+
+Woah! That looks a little scary! We have infinite loops inside of infinite loops, and if statements inside of them. However, this code does exactly the same thing as our first example with a `while` loop above. If Switch2 on our board is ON, LED1 will turn on. And if Switch2 is OFF, LED2 will turn off.
+
+There are a few important things to notice in this example:
+
+1. There is a `break` statement inside of each `if` block. This tells the board that if our conditional is `true`, we want it to "break" out of the current `while` loop and move on. Because each of our inner `while` loops uses this pattern, and is inside an outer `while` loop, the inner loops will switch off between each other.
```

Comment:
I'm hesitant to introduce break here... Do you think it's necessary?

Reply:
I got rid of it